src/leveldb/util/nfs.cc

```diff
@@ -340,5 +346,80 @@ int32_t Nfs::ListDirectory(const std::string& path,
   return 0;
 }
 
+int32_t Nfs::LockDirectory(const std::string& path) {
+  int ret = (*nfsSetDirOwner)(path.c_str());
+  if (ret != 0) {
+    fprintf(stderr, "[LockDirectory] lock dir %s fail, errno: %d",
+        path.c_str(), errno);
+    return -1;
+  }
+
+  std::vector<std::string> lg_and_log;
+  ret = ListDirectory(path, &lg_and_log);
+  if (ret != 0) {
+    fprintf(stderr, "[LockDirectory] list dir %s fail, errno: %d",
+        path.c_str(), errno);
+    return -1;
+  }
+
+  for (size_t i = 0; i < lg_and_log.size(); i++) {
+    if (lg_and_log[i].find(".log") != std::string::npos) {
+      std::string log_file = path + "/" + lg_and_log[i];
+      ret = (*nfsForceRelease)(log_file.c_str());
+      if (ret != 0) {
+        fprintf(stderr, "[LockDirectory] force release file %s fail, errno: %d",
+            log_file.c_str(), errno);
+        return -1;
+      }
+    } else {
+      std::string lg_path = path + "/" + lg_and_log[i];
+      std::vector<std::string> lg_files;
+
+      struct stat fileinfo;
+      ret = (*nfsStat)(lg_path.c_str(), &fileinfo);
+      if (ret != 0) {
+        fprintf(stderr, "[LockDirectory] stat file %s fail, errno: %d",
+            lg_path.c_str(), errno);
+        return -1;
+      }
+
+      if (!S_ISDIR(fileinfo.st_mode)) {
+        continue;
+      }
+
+      ret = (*nfsSetDirOwner)(lg_path.c_str());
+      if (ret != 0) {
```

Comment:
int LockDir(dir) {
	nfsSetDirOwner(dir)
	ListDirectory(dir, children)
	for (i = 0; i < children.size(); i++) {
		if (children[i].find("MANIFEST-") || children[i].find(".log")) {
			nfsForceRelease(dir + "/" + children[i])
		}
	}
}